### PR TITLE
Fix/k8s short circuit if no endpoints

### PR DESCRIPTION
**Pull request by szuecs**
https://github.com/zalando/skipper/issues/1525

### CODE REVIEW

File: dataclients/kubernetes/kube.go [hunk 1]
```diff
@@ -332,6 +332,21 @@ func (c *Client) loadAndConvert() (*clusterState, []*eskip.Route, error) {
 	return state, append(ri, rg...), nil
 }
 
+func shuntRoute(r *eskip.Route) {
+	r.Filters = []*eskip.Filter{
+		{
+			Name: builtin.StatusName,
+			Args: []interface{}{502.0},
```
Comment (AlexanderYastrebov) [inline dataclients/kubernetes/kube.go:339]: 🙈

Comment (szuecs) [reply dataclients/kubernetes/kube.go:339]: :g :D

**Comment by szuecs** [comment]
I need to add or change a few tests that I changed to pass the test build.

**Comment by szuecs** [comment]
:+1:

**Comment by aryszka** [comment]
:+1:

**Pull request merged**